graph/src/components/store/mod.rs

```diff
@@ -244,13 +244,39 @@ impl EntityFilter {
     }
 }
 
+#[derive(Clone, Debug, PartialEq)]
+pub struct EntityOrderByChildObject {
+    pub entity_type: EntityType,
+    pub attribute: Attribute,
+    pub join_attribute: Attribute,
+    pub derived: bool,
```

Comment:
Do we really need the derived here? If the derived field is derived from the parent type, then we can just turn this into an ordinary sort. If it is derived from a grandchild, we shouldn't allow ordering by it because we'd need to join parent, child, and grandchild which we don't do currently.

Reply:
The `derived` value applies to child <-> parent relation. There's of course a case where sorting by `id` could be done without a left join as it "sits" in one of the columns of the parent table. I'm not sure what exactly you mean here, maybe I'm missing something, but sorting rows from `parent` by one of the fields of `child` table requires a left join.